File: lotto-app/modules/ads-environment/ios/AdsEnvironmentModule.swift
import ExpoModulesCore

public class AdsEnvironmentModule: Module {
  public func definition() -> ModuleDefinition {
    Name("AdsEnvironment")

    // TestFlight(샌드박스 영수증) 설치본이면 true → 호출부에서 테스트 광고로 전환.
    // App Store 정식 설치본은 영수증 이름이 "receipt"라 false → 실제 광고.
    Constant("isTestFlight") {
      isRunningInTestFlight()
    }
  }

  private func isRunningInTestFlight() -> Bool {
    guard let receiptURL = Bundle.main.appStoreReceiptURL else { return false }
    return receiptURL.lastPathComponent == "sandboxReceipt"
  }
}

File: lotto-app/modules/ads-environment/android/src/main/java/expo/modules/adsenvironment/AdsEnvironmentModule.kt
package expo.modules.adsenvironment

import expo.modules.kotlin.modules.Module
import expo.modules.kotlin.modules.ModuleDefinition

class AdsEnvironmentModule : Module() {
  override fun definition() = ModuleDefinition {
    Name("AdsEnvironment")

    // Android(Play 내부 테스트 트랙)은 런타임에 신뢰성 있게 판별하기 어려워 기본 false.
    // iOS TestFlight 영수증 판별과 동일한 인터페이스만 맞춰 둔다.
    Constant("isTestFlight") {
      false
    }
  }
}
